### PR TITLE
add: deprecation warning on process.browser

### LINKED ISSUE #4106: Deprecate process.browser
### What problem does this feature solve?

After https://github.com/nuxt/nuxt.js/pull/3910 it'd be convenient to deprecate process.browser and to display a warning on usage (but evaluate it correctly anyway).

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c7965">#c7965</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by destag**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves: #4106 

Added warnings when user uses `process.browser` instead of `process.client`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by pi0** [comment]
@destag Thanks for your PR. But it seems you've used an old fork. Would you please merge latest develop with your changes? (If it went too hard, you can simply create a new branch from dev and re-apply changes)

BTW thanks ❤️

**Comment by destag** [comment]
@pi0 sorry for that. I rebased and pushed new change.

File: packages/app/template/index.js [hunk 1]
```diff
@@ -151,7 +151,10 @@ async function createApp (ssrContext) {
   }
 
   <% if (store) { %>
-  if (process.client) {
+  if (process.client || process.browser) {
+    if (process.browser) {
```
Comment (pi0) [inline packages/app/template/index.js]: Would you please add a `<% if (isDev) { %> ... <% } %>` condition around all `if (browser)` checks? Keeping it, adds a little extra overhead to production builds.

**Comment by pi0** [comment]
Thanks @destag for that. Only a little improvement for production perf. Then should be ready to merge.

**Review by galvez — APPROVED** [review]

**Review by pi0 — APPROVED** [review]

**Pull request merged**